packages/drivers/odsp-driver/src/odspUtils.ts

```diff
@@ -121,14 +123,15 @@ export async function fetchAndParseAsJSONHelper<T>(
         for (const [key, value] of response.headers.entries()) {
             newHeaders.set(key, value);
         }
-        const res = {
+        res = {
             headers: newHeaders,
             content: JSON.parse(text),
         };
-        return res;
     } catch (e) {
         throwOdspNetworkError(`Error while parsing fetch response: ${e}`, fetchIncorrectResponse, response);
```

Comment:
The throwOdspNetworkError  actually also extracts message from response object and then adds to this custom message. This is just to add any custom message so that it is easier to make messages more meaningful as sometimes the error messages from spo are not so meaningful. I think we can trust that the error thrown by spo apis do not contain any PII info.